### PR TITLE
draft gui

### PR DESCRIPTION
**Pull request by LagradOst**

### CODE REVIEW

**Comment by Blatzar** [comment]
Will add ezdl

**Comment by Blatzar** [comment]
Only thing really left is to add eel to imports, should it be some group?

File: anime_downloader/gui/gui.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    print(anime_cls)
+    res = anime_cls.search(query)
+    print(res)
+
+    eel.populate_search([dict(title=i.title, url=i.url, poster=i.poster) for i in res])
+
+@eel.expose
+def get_animeinfo(url):
+    global anime
+    print('getting info of: '+ url)
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/gui.py:30]: `print('Getting information from:')`
Minor grammar and capital letter.

File: anime_downloader/gui/gui.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    anime = get_anime_class(url)(url)
+    print(anime)
+    print(anime[0].source().stream_url)
+    return {
+        'title': anime.title,
+        'length': len(anime),
+    }
+
+def play_mirror_file(player, title, file, ):
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/gui.py:39]: why does this not call anime-dl's mpv function, that basically does the Popen along with some other stuff.

File: anime_downloader/gui/gui.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        # Fix util-prase episode range
+        external_downloader = Config['dl']['external_downloader']
+        file_format = Config['dl']['file_format']
+        download_dir = Config['dl']['download_dir']
+        speed_limit= Config['dl']['speed_limit']
+        util.external_download(external_downloader, i, file_format,path=download_dir, speed_limit=speed_limit)
+
+@eel.expose
+def load_vlc(start, end): 
+    #print(anime)
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/gui.py:66]: should probably remove these once you are happy with this function.

File: anime_downloader/gui/gui.py (hunk 2 above)
Comment (Blatzar) [reply anime_downloader/gui/gui.py:39]: Will fix, my bro was unaware of the utils in the project

File: anime_downloader/gui/gui.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    #test = AnimeEpisode(episodes[0][1], anime['quality'], anime['_fallback_qualities'])
+    episodes = util.parse_ep_str(anime, start+':'+end)
+    text = "#EXTM3U\n"
+    for i in episodes:
+        print(i.__dict__)
+
+    for i in episodes:
+        text += f"#EXTINF:,Episode {(i.ep_no)}\n" 
+        text += i.source().stream_url + "\n"
+    text_file = open("MirrorList.m3u8", "w")
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/gui.py:77]: Isn't this gonna save the file in cwd? Is that what you want?

File: anime_downloader/gui/gui.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    text_file.write(text)
+    text_file.close()
+    print("Playing m3u8 file")
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/gui.py:80]: fullstop

File: anime_downloader/gui/gui.py (hunk 4 above)
Comment (Blatzar) [reply anime_downloader/gui/gui.py:77]: Will fix to the config

File: anime_downloader/gui/web/main.css [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+    color: #fff;
+    background-color: #303F9F;
+    border-color: transparent;
+}
+
+    .btn-primary:hover {
+        color: #fff;
+        background-color: #3c50d3;
+        border-color: #fff;
+    }*/
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/web/main.css:567]: These are all commented out, right? Are you going to actually use these?

File: anime_downloader/gui/web/main.html [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    function StartLoad() {
+        document.getElementById('ChangeItems').style = "width:100%; margin-top:0px; height:2px; opacity:1;";
+    }
+
+    function StopLoad() {
+        document.getElementById('ChangeItems').style = "width:100%; margin-top:0px; height:2px; opacity:0;";
+    }
+    function js_load_vlc() {
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/web/main.html:157]: non-consistent function spacing

File: anime_downloader/gui/web/main.html [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+
+
+    async function populate_animeinfo(url) {
+        StartLoad();
+        let info = await eel.get_animeinfo(url)();
+        StopLoad();
+        console.log(info)
+        on();
+        document.getElementById('anime-title').innerHTML = info.title;
+        //  c = info.anime;
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/web/main.html:172]: Another commented out line of code, are you going to use this?

File: anime_downloader/gui/gui.py (hunk 5 above)
Comment (Blatzar) [reply anime_downloader/gui/gui.py:80]: Please elaborate

File: anime_downloader/gui/web/main.html [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+                }
+                this.classList.add('has-background-primary')
+                //this.classList.add('has-text-light')
+                console.log(this.url);
+                populate_animeinfo(this.url);
+            })
+            search_res.appendChild(ele);
+        }
+
+        /* for(var i = 0; i < results.length; i++) {
```
Comment (MichaelDavidJohnson) [inline anime_downloader/gui/web/main.html:203]: Commented out code, are you going to use this?

File: anime_downloader/gui/gui.py (hunk 5 above)
Comment (MichaelDavidJohnson) [reply anime_downloader/gui/gui.py:80]: Fullstops follow the end of sentences.

Comment (MichaelDavidJohnson) [reply anime_downloader/gui/gui.py:80]: https://en.wikipedia.org/wiki/Full_stop

**Comment by Blatzar** [comment]
Accidentally marked for review too early since it worked, all of the minor details as comments will be fixed

**Comment by MichaelDavidJohnson** [comment]
Yeah, it's definitely hard to review the code when most of it's commented out... I don't know if it's commented out for future implementation or if they actually need to be gone.

**Comment by LagradOst** [comment]
I just comment out stuff instead of removing it most of the time, old habit because I most of the time work alone. Will remove everything that is bloat before merge or review.

**Comment by LagradOst** [comment]
I just comment out stuff instead of removing it most of the time, old habit because I most of the time work alone. Will remove everything that is bloat before merge or review.

**Comment by LagradOst** [comment]
I just comment out stuff instead of removing it most of the time, old habit because I most of the time work alone. Will remove all bloat before merge or review.

File: anime_downloader/gui/gui.py (hunk 1 above)
Comment (vn-ki) [reply anime_downloader/gui/gui.py:30]: Use `click.echo` instead of print

File: anime_downloader/gui/gui.py [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    providers = [i[1] for i in ALL_ANIME_SITES]
+    eel.populate_dropdown(providers)
+
+
+@eel.expose
+def anime_search(query, provider):
+    print('serching')
+    print(query, provider)
+    anime_cls = get_anime_class(provider)
+    print(anime_cls)
```
Comment (vn-ki) [inline anime_downloader/gui/gui.py:21]: `log.debug`?

File: anime_downloader/gui/gui.py [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+def play_mirror_file(player, title, file, ):
+    if player == 'mpv':
+        p = subprocess.Popen([
+            player,
+            '--title={}'.format(title),
+            file,
+            ])
+    else:
+        p = subprocess.Popen([ player, file
+            ])
```
Comment (vn-ki) [inline anime_downloader/gui/gui.py:48]: This is weird formatting. Also either use the mpv class from the `players` or use the one from util. I prefer former.

File: anime_downloader/gui/gui.py [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    for i in episodes:
+        # Fix util-prase episode range
+        external_downloader = Config['dl']['external_downloader']
+        file_format = Config['dl']['file_format']
+        download_dir = Config['dl']['download_dir']
+        speed_limit= Config['dl']['speed_limit']
+        util.external_download(external_downloader, i, file_format,path=download_dir, speed_limit=speed_limit)
+
+@eel.expose
+def load_vlc(start, end):
```
Comment (vn-ki) [inline anime_downloader/gui/gui.py:65]: What does this function do?

Comment (LagradOst) [reply anime_downloader/gui/gui.py:65]: load_vlc, starts mpv should rename the function. (This pull is not ready for review yet, and I will fix all small things before review or merge)

**Comment by AbdullahM0hamed** [comment]
Will you still work on this? Or drop it in favour of #526?

**Comment by vn-ki** [comment]
Closing in favour of #526

**Comment by vn-ki** [comment]
Oh, I thought this was mine. Lagrad can take this decision. Re-opening.

**Comment by LagradOst** [comment]
> Will you still work on this? Or drop it in favour of #526? 

I will work on this when cs2 electron is released. I dont have all the time in the world to work on stuff

**Comment by LagradOst** [comment]
(I will not work on the qt version)

**Comment by IguanasInPyjamas** [comment]
Seems like QT may be the way forward and then when you finally can get around to it you can replace it? QT is ready to ship now pretty much. Seems like a waste to wait for something that isn't really been worked on

**Comment by LagradOst** [comment]
> Seems like QT may be the way forward and then when you finally can get around to it you can replace it? QT is ready to ship now pretty much. Seems like a waste to wait for something that isn't really being worked on

you dont need to wait for me to finish this, this project is only a side project that I _might_ work on for my own use

**Comment by IguanasInPyjamas** [comment]
Should I close the PR then? Makes the project cleaner if I remove a draft pr that may never be worked on again

**Pull request closed without merging**